instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb

```diff
@@ -0,0 +1,155 @@
+          end
+
+          def lru_cache
+            # When SQL is being sanitized, we know that this cache will
+            # never be more than 50 entries * 2000 characters (so, presumably
+            # 100k bytes - or 97k). When not sanitizing SQL, then this cache
+            # could grow much larger - but the small cache size should otherwise
+            # help contain memory growth. The intended use here is to cache
+            # prepared SQL statements, so that we can attach a reasonable
+            # `db.sql.statement` value to spans when those prepared statements
+            # are executed later on.
+            @lru_cache ||= ::LruRedux::ThreadSafeCache.new(50)
+          end
+
+          # Rubocop is complaining about 19.31/18 for Metrics/AbcSize.
+          # But, getting that metric in line would force us over the
+          # module size limit! We can't win here unless we want to start
+          # abstracting things into a million pieces.
+          def span_attrs(kind, *args) # rubocop:disable Metrics/AbcSize
+            if kind == :query
+              operation = extract_operation(args[0])
+              sql = obfuscate_sql(args[0])
+            else
+              statement_name = args[0]
+
+              if kind == :prepare
+                sql = obfuscate_sql(args[1])
+                lru_cache[statement_name] = sql
+                operation = 'PREPARE'
+              else
+                sql = lru_cache[statement_name]
+                operation = 'EXECUTE'
+              end
+            end
+
+            attrs = { 'db.operation' => validated_operation(operation), 'db.statement' => sql, 'db.postgresql.prepared_statement_name' => statement_name }
+            attrs.reject! { |_, v| v.nil? }
+
+            [span_name(operation), client_attributes.merge(attrs)]
+          end
+
+          def extract_operation(sql)
+            # From: https://github.com/open-telemetry/opentelemetry-js-contrib/blob/9244a08a8d014afe26b82b91cf86e407c2599d73/plugins/node/opentelemetry-instrumentation-pg/src/utils.ts#L35
+            sql.to_s.split[0].to_s.upcase
+          end
+
+          def span_name(operation)
+            [validated_operation(operation), database_name].compact.join(' ')
+          end
+
+          def validated_operation(operation)
+            operation if PG::Constants::SQL_COMMANDS.include?(operation)
+          end
+
+          def obfuscate_sql(sql)
+            return sql unless config[:enable_sql_obfuscation]
+
+            # Borrowed from opentelemetry-instrumentation-mysql2
+            return 'SQL query too large to remove sensitive data ...' if sql.size > 2000
+
+            # From:
+            # https://github.com/newrelic/newrelic-ruby-agent/blob/9787095d4b5b2d8fcaf2fdbd964ed07c731a8b6b/lib/new_relic/agent/database/obfuscator.rb
+            # https://github.com/newrelic/newrelic-ruby-agent/blob/9787095d4b5b2d8fcaf2fdbd964ed07c731a8b6b/lib/new_relic/agent/database/obfuscation_helpers.rb
```

Comment:
👋 Hey @mwlang hope all is well. At the SIG today we were chatting on how useful this obfuscation code is for orm/db instrumentation in ruby. It would be wonderful to be able to re-use the Obfuscation Module in opentelemetry-ruby directly. Would you be open to making a contribution to add this if you have bandwidth? And if not, any feedback or guidance on this approach?